src/commands/pickFuncProcess.ts

```diff
@@ -10,44 +10,54 @@ import * as vscode from 'vscode';
 import { hostStartTaskName } from '../constants';
 import { preDebugValidate, type IPreDebugValidateResult } from '../debug/validatePreDebug';
 import { ext } from '../extensionVariables';
-import { getFuncPortFromTaskOrProject, isFuncHostTask, runningFuncTaskMap, stopFuncTaskIfRunning, type IRunningFuncTask } from '../funcCoreTools/funcHostTask';
+import { AzureFunctionTaskDefinition, getFuncPortFromTaskOrProject, isFuncHostTask, runningFuncTaskMap, stopFuncTaskIfRunning, type IRunningFuncTask } from '../funcCoreTools/funcHostTask';
 import { localize } from '../localize';
 import { delay } from '../utils/delay';
 import { requestUtils } from '../utils/requestUtils';
 import { taskUtils } from '../utils/taskUtils';
 import { getWindowsProcessTree, ProcessDataFlag, type IProcessInfo, type IWindowsProcessTree } from '../utils/windowsProcessTree';
 import { getWorkspaceSetting } from '../vsCodeConfig/settings';
 
-const funcTaskReadyEmitter = new vscode.EventEmitter<vscode.WorkspaceFolder>();
+const funcTaskReadyEmitter = new vscode.EventEmitter<string>();
 export const onDotnetFuncTaskReady = funcTaskReadyEmitter.event;
 
 export async function startFuncProcessFromApi(
-    workspaceFolder: vscode.WorkspaceFolder,
     buildPath: string,
-    args?: string[]
+    args: string[],
+    env: { [key: string]: string }
 ): Promise<{ processId: string; success: boolean; error: string }> {
     const result = {
         processId: '',
         success: false,
         error: ''
     };
 
+    const uriFile: vscode.Uri = vscode.Uri.file(buildPath)
+
+    const azFuncTaskDefinition: AzureFunctionTaskDefinition = {
+        // VS Code will only run a single instance of a task `type`,
+        // the path will be used here to make each project be unique.
+        type: `func ${uriFile.fsPath}`,
+        functionsApp: uriFile.fsPath
+    }
+
     let funcHostStartCmd: string = 'func host start';
     if (args) {
         funcHostStartCmd += ` ${args.join(' ')}`;
     }
 
     await callWithTelemetryAndErrorHandling('azureFunctions.api.startFuncProcess', async (context: IActionContext) => {
         try {
-            await waitForPrevFuncTaskToStop(workspaceFolder);
-            const funcTask = new vscode.Task({ type: 'func' },
-                workspaceFolder,
+            await waitForPrevFuncTaskToStop(azFuncTaskDefinition.functionsApp);
+            const funcTask = new vscode.Task(azFuncTaskDefinition,
+                vscode.TaskScope.Global,
```

Comment:
What does `vscode.TaskScope.Global` help do here?

Reply:
Technically its unused, but its a required parameter for the VS Code API. I was hoping this would let it know that this task is used globally and not just for a specific workspace. 

https://github.com/microsoft/vscode/blob/c264ebed9425f3569ab5eb2c6fbc6a6bb4abaaaf/src/vscode-dts/vscode.d.ts#L8726C1-L8738C244